src/EditorFeatures/CSharpTest/CodeActions/MoveType/MoveTypeTests.RenameFileOrType.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeActions.MoveType
+{
+    public partial class MoveTypeTests : CSharpMoveTypeTestsBase
+    {
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveType)]
+        public async Task SingleClassInFile()
+        {
+            var code =
+@"[||]class Class1 { }";
+
+            var expectedDocumentName = "Class1.cs";
+
+            await TestRenameFileToMatchTypeAsync(code, expectedDocumentName);
+
+            var codeWithTypeRenamedToMatchFileName =
```

Comment:
agreed. thought I could shove it here and test more instead of writing more tests. I now think more tests where each of them test a dedicated scenario is good.